codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/parse/JsonParserGenerator.kt

```diff
@@ -197,31 +183,12 @@ class JsonParserGenerator(
         )
     }
 
-    fun renderStructure(
-        writer: RustWriter,
+    fun serverStructureParser(
```

Comment:
Should we just call this `serverParser` and add it to the StructuredDataParserGenerator interface (can have `TODO()` in the other generators for now)?